ext/compressors/iaa/iaa_compress.c

```diff
@@ -0,0 +1,208 @@
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+/*
+ * We need to include the configuration file to detect whether this extension is being built into
+ * the WiredTiger library; application-loaded compression functions won't need it.
+ */
+#include <wiredtiger_config.h>
+
+#include <wiredtiger.h>
+#include <wiredtiger_ext.h>
+#include <time.h>
+#include "iaaInterface-c.h"
+
+
+#ifdef _MSC_VER
+#define inline __inline
+#endif
+
+/* Local compressor structure. */
+typedef struct {
+    WT_COMPRESSOR compressor; /* Must come first */
+
+    WT_EXTENSION_API *wt_api; /* Extension API */
+} iaa_COMPRESSOR;
+
+/*
+ * iaa_error --
+ *     Output an error message, and return a standard error code.
+ */
+static int
+iaa_error(WT_COMPRESSOR *compressor, WT_SESSION *session, const char *call, int error)
+{
+    WT_EXTENSION_API *wt_api;
+
+    wt_api = ((iaa_COMPRESSOR *)compressor)->wt_api;
+
+    (void)wt_api->err_printf(wt_api, session, "iaa error: %s: %u", call, error);
+    return (WT_ERROR);
+}
+
+/*
+ * iaa_compress --
+ *     WiredTiger iaa compression.
+ */
+static int
+iaa_compress(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, size_t src_len,
+  uint8_t *dst, size_t dst_len, size_t *result_lenp, int *compression_failed)
+{
+    uint32_t result_len = doCompressData(compressor, session, src, (uint32_t)src_len, dst, (uint32_t)dst_len);
+    if(result_len > 0) {
+        *compression_failed = 0;
+        *result_lenp = (size_t)result_len;
+        return (0);
+    }
+    *compression_failed = 1;
+    return (iaa_error(compressor, session, "iaa_compress", -1));
+}
+
+/*
+ * iaa_decompress --
+ *     WiredTiger iaa decompression.
+ */
+static int
+iaa_decompress(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, size_t src_len,
+  uint8_t *dst, size_t dst_len, size_t *result_lenp)
+{
+    uint32_t result_len = 0;
+    int ret;
+    doDecompressData(compressor, session, src, (uint32_t)src_len, dst, (uint32_t)dst_len, &result_len);
+    if(result_len > 0) {
+        *result_lenp = (size_t)result_len;
+        ret = 0;
+    } else {
+        ret = -1;
+    }
+    return (ret == 0 ? 0 : iaa_error(compressor, session, "iaa_decompress", ret));
+}
+
+/*
+ * iaa_terminate --
+ *     WiredTiger iaa compression termination.
+ */
+static int
+iaa_terminate(WT_COMPRESSOR *compressor, WT_SESSION *session)
+{
+    (void)session; /* Unused parameters */
+
+    free(compressor);
+    return (0);
+}
+
+/*
+ * iaa_pre_size --
+ *     WiredTiger iaa destination buffer sizing.
+ */
+static int
+iaa_pre_size(
+  WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, size_t src_len, size_t *result_lenp)
+{
+    (void)compressor; /* Unused parameters */
+    (void)session;
+    (void)src;
+
+    /*
+     * iaa requires the dest buffer be somewhat larger than the source. Fortunately, this is fast
```

Comment:
Is this comment still valid for iaa?

Reply:
it is invalid, we will update it as below:
    /*
     * Get the upper-bound of the buffer size needed by the compression.
     */
    *result_lenp = getMaxCompressedDataSize((uint32_t)src_len);;
    return (0);